drivers/platform/aducm3029/irq.c

```diff
@@ -0,0 +1,298 @@
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include "irq.h"
+#include "irq_extra.h"
+#include "error.h"
+#include <stdlib.h>
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+/** The number of the first external interrupt, used by NVIC */
+#define BASE_XINT_NB			(XINT_EVT0_IRQn)
+
+/** Number of interrupts controllers available */
+#define NB_INTERRUPT_CONTROLLERS	1u
+
+/** Map the interrupt ID to the ADI_XINT_EVENT associated event */
+static const uint32_t id_map_event[NB_EXT_INTERRUPTS] = {
+	ADI_XINT_EVENT_INT0, // ID 0
+	ADI_XINT_EVENT_INT1, // ID 1
+	ADI_XINT_EVENT_INT2, // ID 2
+	ADI_XINT_EVENT_INT3  // ID 3
+};
+
+/******************************************************************************/
+/***************************** Global Variables *******************************/
+/******************************************************************************/
+
+/** Bitmap with initialized interrupt */
+static uint32_t		initialized;
```

Comment:
Maybe move this inside the descriptor?

Reply:
It is used to store if the irq controller was already initialized and to not let the user create multiple instances of the controller.